### PR TITLE
Division by 10

### PR DESCRIPTION
**Pull request by raulgarfer**
Function that calculates the division by 10 from a number between 0 and 255. Returns an entire number.

### CODE REVIEW

File: cpctelera/src/math/cpct_divideby10.asm [hunk 1]
```diff
@@ -0,0 +1,176 @@
+;;-----------------------------LICENSE NOTICE------------------------------------
+;;  This file is part of CPCtelera: An Amstrad CPC Game Engine 
+;;
+;;  Created by Nesu 2022. Optimized by Joaquín Ferrero & Nesu in 2023
+;;  Copyright (C) 2021 Joaquín Ferrero (https://github.com/joaquinferrero)
+;;  Copyright (C) 2021 Nestornillo (https://github.com/nestornillo)
+;;  Copyright (C) 2024 raulgarfer (https://github.com/raulgarfer)
+;;  Copyright (C) 2024 cpcitor (https://github.com/fidergo-stephane-gourichon/)
```
Comment (cpcitor) [inline cpctelera/src/math/cpct_divideby10.asm]: This URL is more relevant: https://github.com/cpcitor/

File: cpctelera/src/math/cpct_divideby10.asm [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+;; 3) (1/8 + 1/2 + 1/1 ) / 8
+;; Previously, we omitted N/2; let's insert it now:
+;; 4) N/2 * 1 / 5 is similar to
+;; 5) N/2 * (1/8  + 1/2 + 1/1) / 8, which refactors to
+;; 6)       (N/16 + N/4 + N/2) / 8
+;; Taking the example of number 125, perform the operations:
+;; 1) (125/16 + 125/4 + 125/2) / 8
+;; 2) (   7   +   31  +   62  ) / 8
+;; 3) (           99          ) / 8
+;; 4)             12
```
Comment (cpcitor) [inline cpctelera/src/math/cpct_divideby10.asm]: This explanation is nice as a simplified explanation and can IMHO be kept.
Yet it is not exactly what the routine computes.
The explanation corresponds to comments on lines below where it is said that doing this yields too small results: https://github.com/lronaldo/cpctelera/blob/ad99adb6926f4cc79ec1d679c3dfb42b684dc6f6/cpctelera/src/math/cpct_divideby10.asm#L120-L123
All this is okay, perhaps just add a sentence like "This is a simplified explanation, the code actually groups partial results together so as to minimize the number of shift-right operations, and it happens that the result is more precise that way."
Or something like that.

**Review by cpcitor — COMMENTED** [review]
I can't review all the nice properties that @lronaldo enumerated in the previous PR, just making comments about things I noticed. Cheers everyone!

File: cpctelera/src/math/cpct_divideby10.asm (hunk 2 above)
Comment (raulgarfer) [reply cpctelera/src/math/cpct_divideby10.asm]: Added the explanation, improved with your comments.

**Comment by raulgarfer** [comment]
I have formatted the description. adden cpcitor as indicated. added explanation . our friend nestor said me that c bindings are wrong. my level of c is very low at this time.
i wish that someone could fix this point.

**Review by raulgarfer — COMMENTED** [review]
thanks for your cooperation

**Review by raulgarfer — COMMENTED** [review]
better explanations

**Pull request closed without merging**